Expose hud_oldscale and hud_scalefactor to the menu; add mGrayCheckVal

jekyllgrim opened the pull request:
This adds two separate changes, but since the menu one is mechanically related to the one to OptionMenuItem, they have to go together:
1. Adds the `graycheckVal` argument to OptionMenu items, and the corresponding `mGrayCheckVal` field to OptionMenuItem. This will allow users to specify an explicit value for the graycheck cvar, so that items can be grayed-out not only when it's 0/false. It's a bit awkward because in some elements `graycheckVal` follows `graycheck` directly, but in others the `centered` argument is in the middle, but there's nothing I can do about it and I think it's better to have it this way than not at all.
2. Exposes `hud_oldscale` and `hud_scalefactor` to the menu (with corresponding new entries to LANGUAGE.csv). With the help of the new `graycheckval`, when `hud_oldscale` is disabled, the traditional HUD scale sliders become grayed-out. `hud_scalefactor` is exposed with 0.025 increments.

https://github.com/ZDoom/gzdoom/assets/42670463/037accf2-ff04-4616-8654-cdbf82f8c99e

There are no changes to the default values, so this shouldn't lead to any confusion among the users.

P.S. Please don't mind that in the video the scalefactor slider goes up to 2.000; in the commit it's limited to 0.0-1.0. Although ideally I *would* like the ability to make it go higher, but I don't know how to do that.

coelckers reviewed (changes requested):
Please do not change the language.csv file. The text needs to go into the Google spreadsheet. This will be overwritten when we next do a text file update.

coelckers commented:
Another thing here is that it is rather pointless to make menu layout changes. VkDoom has completely revamped the menu layout and this would get lost there as well.

jekyllgrim commented:
> Another thing here is that it is rather pointless to make menu layout changes. VkDoom has completely revamped the menu layout and this would get lost there as well.

I'm not quote sure what to do with this information. How else should this be exposed to the menus?

coelckers commented:
What this means is that much of the current menus will just be tossed out. So I'm a bit careful with adding new content.

jekyllgrim commented:
Well, I'd rather keep the menu change for now, since I see no harm in doing that even if it gets purged later (and I assume scalefactor will still need to be exposed in the new menus in some way, however it's done).

I can revert the LANGUAGE change (I expected I might need to). Can I get write access to jekyllgrim@gmail.com to be able to add those strings in the Google sheet?

jekyllgrim commented:
The necessary LANGUAGE lines have been added to the Google sheet.

coelckers commented:
Please resolve the conflicts so this can be merged.

jekyllgrim commented:
> Please resolve the conflicts so this can be merged.

I'm not sure what they are, exactly? I see "changes requested", but as requested, I've reverted the commit to LANGUAGE and added the necessary strings to the Google Sheet instead. Doesn't seem like I can do anything else on my end.

madame-rachelle commented:
It seems like this can be merged ok with a squash merge or a standard history convergence merge, but for some reason it doesn't like a rebase merge.

jekyllgrim commented:
Okay, I squashed all 3 commits together, and now language.csv should be completely untouched. Hopefully that works now :)

madame-rachelle commented:
Looks like all is good now - about the menu -

VkDoom may be revamping the menu, but the change is not yet GZDoom. Therefore, it's probably best to simply accept the menu change, and if VkDoom gets it downstream they will deal with it.

nashmuhandes commented:
This should just be merged into GZDoom. I'll deal with the menu delta on VKDoom's side.
